checker/check_runner.go

```diff
@@ -39,15 +39,56 @@ type CheckFn func(*CheckRequest) CheckResult
 
 type CheckNameToFnMap map[string]CheckFn
 
+// Types of details.
+type DetailType int
+
+const (
+	DetailInfo DetailType = iota
+	DetailWarn
+	DetailDebug
+)
+
+// CheckDetail contains information for each detail.
+//nolint:govet
+type CheckDetail struct {
+	Type DetailType // Any of DetailWarn, DetailInfo, DetailDebug.
+	Msg  string     // A short string explaining why the details was recorded/logged..
+}
+
+type DetailLogger interface {
+	Info(desc string, args ...interface{})
```

Comment:
How about a fn signature like this:

```
Log(detail CheckDetail)
```

Since `CheckDetail` already has `Info/Warn/Debug`, `DetailLogger` can internally figure out how to log these.

Reply:
Nvm, I misunderstood this fn.